src/Illuminate/Testing/TestJson.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+namespace Illuminate\Testing;
+
+use Illuminate\Contracts\Support\Jsonable;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use Illuminate\Testing\Assert as PHPUnit;
+use JsonSerializable;
+
+class TestJson
+{
+    /**
+     * The original encoded json.
+     *
+     * @var \Illuminate\Contracts\Support\Jsonable|\JsonSerializable|array
+     */
+    public $json;
+
+    /**
+     * The decoded json contents.
+     *
+     * @var array|null
+     */
+    protected $decoded;
+
+    /**
+     * Create a new test json instance.
+     *
+     * @param  \Illuminate\Contracts\Support\Jsonable|array  $jsonable
+     * @return void
+     */
+    public function __construct($jsonable)
+    {
+        $this->json = $jsonable;
+
+        if ($jsonable instanceof JsonSerializable) {
+            $this->decoded = $jsonable->jsonSerialize();
+        } elseif ($jsonable instanceof Jsonable) {
+            $this->decoded = json_decode($jsonable->toJson(), true);
+        } elseif (is_array($jsonable)) {
+            $this->decoded = $jsonable;
+        } else {
+            $this->decoded = json_decode($jsonable, true);
```

Comment:
When is this called? The phpdoc doesn't allow strings.

Reply:
I need to update the phpdoc, should be `Illuminate\Contracts\Support\Jsonable|JsonSerializable|string` 

string usually come from `TestResponse::getContent()`